### ISSUE TITLE
WarioWare: Twisted! Gyroscope support

### ISSUE THREAD

**Issue by konistehrad** [jasarien/Provenance#672]
Evening, all! Recently I fired up WarioWare: Twisted and discovered the gyro support in the game wasn't supported by PVGBA. I believe I see the change needed in PVGBA's `RTC.cpp` file (gba4ios shows [a correct implementation](https://bitbucket.org/rileytestut/gba4ios/src/f7a84d8be19ba16eab5767e686120b1625e8277f/emu-ex-plus-alpha/GBA.emu/src/vbam/gba/RTC.cpp?at=master&fileviewer=file-view-default#RTC.cpp-51)), but before opening a PR I wanted to talk about the UX implications of futzing with rotation in relation to UI orientation.

My first thought was to conditionally lock rotation in-game if it was detected that a gyroscope-enabled game was being played, but without feedback I worry about user confusion ("wait, why isn't this changing?") Any additional feedback on this would be much appreciated; there may be an implementation for a platform that I don't have experience with even!

**Comment by JoeMatt** [comment]
Makes sense to me, what you could do is add a public method to the `PVEmulatorCore` class like
`@property (nonatomic, assign) BOOL isUsingGyroscope;` that defaults to NO;

Then in `PVEmulatorViewController.m`, override 
```
- (BOOL)shouldAutorotate {
     // Always rotate when menu open, otherwise always rotate unless isUsingGyroscope has been activated
    return !self.emulatorCore.isUsingGyroscope || self.isShowingMenu;
}
```

Then finally in `PVGBAEmulatorCore`, when you detect this rom in `loadFileAtPath`, set `self.isUsingGyroscope = YES;`

I'm pretty sure that would work. As far as the alert, that looks OK to me.

**Comment by JoeMatt** [comment]
Closing: WarioWare: Twisted! gyroscope support is extremely hardware-specific (original cartridge gyro chip) and not exposed by the VBA-M core. Not currently feasible.
